wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/StructJsonAdapter.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+
+/**
+ * Json adapter for structs, list values, values, or null values as defined in
+ * `google/protobuf/struct.proto`.
+ */
+internal object StructJsonAdapter : JsonAdapter<Any>() {
+  override fun toJson(out: JsonWriter, any: Any?) {
+    when (any) {
+      null -> out.nullValue()
+      is Number -> out.value(any)
+      is String -> out.value(any)
+      is Boolean -> out.value(any)
+      is Map<*, *> -> {
+        if (any.isEmpty()) {
+          skip(out)
+          return
+        }
+
+        out.beginObject()
+        for ((key, value) in any) {
+          out.name(key as String)
+          toJson(out, value)
+        }
+        out.endObject()
+      }
+      is List<*> -> {
+        if (any.isEmpty()) {
+          skip(out)
+          return
+        }
+
+        out.beginArray()
+        for (value in any) {
+          toJson(out, value)
+        }
+        out.endArray()
+      }
+      else -> throw IllegalArgumentException("unexpected struct value: $any")
+    }
+  }
+
+  override fun fromJson(input: JsonReader): Any? {
+    return input.readJsonValue()
+  }
+
+  private fun skip(out: JsonWriter) {
+    val serializeNulls: Boolean = out.serializeNulls
+    out.serializeNulls = false
+    try {
+      out.nullValue()
```

Comment:
This is wrong. If the List is a member of another List, you want `[]`, not `null`.

Reply:
You're correct, thank you. I fixed it in 8bec13f